### PR TITLE
Add new parameters for resize and alloc

### PR DESCRIPTION
**Pull request by SamTebbs33**
Zig update.﻿

### CODE REVIEW

File: src/kernel/filesystem/vfs.zig [hunk 1]
```diff
@@ -560,8 +561,8 @@ test "mount" {
 
     // The root fs
     var testfs = try testInitFs(allocator);
-    defer testfs.deinit();
     defer allocator.destroy(testfs);
+    defer testfs.deinit();
```
Comment (DrDeano) [inline src/kernel/filesystem/vfs.zig:565]: I think it still needs to be `deinit` then `free` as when freeing first, will be deiniting a free pointer

File: src/kernel/filesystem/vfs.zig [hunk 2]
```diff
@@ -594,8 +595,8 @@ test "mount" {
 test "traversePath" {
     var allocator = testing.allocator;
     var testfs = try testInitFs(allocator);
-    defer testfs.deinit();
     defer allocator.destroy(testfs);
+    defer testfs.deinit();
```
Comment (DrDeano) [inline src/kernel/filesystem/vfs.zig:599]: Same here (5 in total)

File: src/kernel/heap.zig [hunk 3]
```diff
@@ -182,16 +182,18 @@ const FreeListAllocator = struct {
     /// Arguments:
     ///     IN allocator: *std.Allocator - The allocator to resize within.
     ///     IN old_mem: []u8 - The buffer to resize.
+    ///     IN mem_align: u29 - The original alignment for old_mem.
```
Comment (DrDeano) [inline src/kernel/heap.zig]: Maybe `old_align` would be better

File: src/kernel/filesystem/vfs.zig (hunk 1 above)
Comment (SamTebbs33) [reply src/kernel/filesystem/vfs.zig:565]: I was getting seg faults when using the previous order and upon looking at the docs I read that the deferred statements are run in reverse order. which is why I defer deinitting the testfs last.

File: src/kernel/heap.zig (hunk 3 above)
Comment (SamTebbs33) [reply src/kernel/heap.zig]: Sure. This name was copied straight from std lib.

File: src/kernel/filesystem/vfs.zig (hunk 1 above)
Comment (DrDeano) [reply src/kernel/filesystem/vfs.zig:565]: Ah, ok, I thought it was the other way, so I will have to re-order some of mine then

Comment (SamTebbs33) [reply src/kernel/filesystem/vfs.zig:565]: Yeah me too! It actually makes sense to have it reversed though.

**Review by DrDeano — APPROVED** [review]
:)

**Pull request merged**